pkg/scheduler/preemption/preemption.go

```diff
@@ -116,6 +116,14 @@ func (p *Preemptor) GetTargets(wl workload.Info, assignment flavorassigner.Assig
 		}
 		return minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, true, allowBorrowingBelowPriority)
 	}
+
+	// the target queue is exhausted in all requested flavors, so preempting
+	// workloads from other cluster queues in the cohort is no help. so we
+	// only preempt workloads from the target cluster queue.
+	if isQueueExhaustedForAllRequestedFlavors(&wl, assignment, cq) {
```

Comment:
could we instead pass this flag to `findCandidates`? I guess no, because we still need to allow borrowWithinCohort when enabled, right?

Reply:
Right, but not only that, this is an optimization only if `allowBorrowing=false`. If `allowBorrowing=true` (as in a couple of other branches) we still need to include candidates from other cluster queues. Even if the target CQ is exhausted, because preempting workloads from other CQs may help.

Reply:
So, this is essentially only to skip the check below `targets := minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, false, nil)`. Maybe we can organize the code differently to make it cleaner.